net2/http2/load_balanced_pool.go

```diff
@@ -143,6 +150,10 @@ func (pool *LoadBalancedPool) Do(req *http.Request) (*http.Response, error) {
 			// protection when service may still be up but have higher rate of
 			// 500s for whatever reason.
 			pool.markInstanceDown(idx, instance, time.Now().Add(markDownDuration).Unix())
+		} else if isDown {
+			// If an instance was marked as down, but succeeded, reset the mark down timer, so
+			// instance is treated as healthy right away.
+			pool.markInstanceDown(idx, instance, 0)
```

Comment:
using pool.markInstanceDown is misleading.  Maybe wrap this in a helper function pool.markInstanceUp.  LG otherwise